tapchannelmsg/records.go

```diff
@@ -736,6 +815,358 @@ func DecodeHtlcAuxLeaf(blob tlv.Blob) (*HtlcAuxLeaf, error) {
 	return &h, nil
 }
 
+// AssetSig is a record that represents the signature for spending an asset
+// output.
+type AssetSig struct {
+	// AssetID is the asset ID that the signature is for.
+	AssetID tlv.RecordT[tlv.TlvType0, asset.ID]
+
+	// Sig is the signature for the asset spend.
+	Sig tlv.RecordT[tlv.TlvType1, lnwire.Sig]
+
+	// SigHashType is the sigHash type that was used to create the
+	// signature.
+	SigHashType tlv.RecordT[tlv.TlvType2, uint32]
```

Comment:
Use `txscript.SigHashType` here instead of `uint32`?

Reply:
This would lead to `panic: unknown primitive type: *txscript.SigHashType [recovered]`. We either need to have types that have a `Record()` method or are from a list of defined primitive types (no type aliases allowed).